app/controllers/concerns/foreman/controller/registration.rb

```diff
@@ -18,6 +18,7 @@ def global_registration_vars
     location = Location.authorized(:view_locations).find(params['location_id']) if params['location_id'].present?
     host_group = Hostgroup.authorized(:view_hostgroups).find(params['hostgroup_id']) if params["hostgroup_id"].present?
     operatingsystem = Operatingsystem.authorized(:view_operatingsystems).find(params['operatingsystem_id']) if params["operatingsystem_id"].present?
+    ssl_ca_file = File.read(Setting[:ssl_ca_file]) rescue nil
```

Comment:
also should we hard fail in the case the file can't be read? this effectively breaks the registration process, curl will fail for sure, so it's better to fail earlier and provide explanatory error message

Reply:
But what if somebody don't want to use https? There can be users like that.

Reply:
I believe we need some error handling here, maybe with a macro?

Reply:
I think we should ignore the error only in the case if file can't be read and the current request is not https. If it is https, we should consider this as a hard failure. All the links we generate using path helpers will inherit the protocol too.

> I believe we need some error handling here, maybe with a macro?

do we need macros if the endpoint fails even before the template is rendered? or could you describe a bit better what you have in mind?

Reply:
I believe we should use the chance to "upgrade" the connection to https, I would not spend any time to make this intelligently work with both HTTP and HTTPs.

What I meant with error handling is: If the file can't be read, we need to give the user a clear error message that this failed and why it failed. And my idea was, that we could actually make this a macro (that would print the certificate). And only if the macro is used (it could come in handy during normal provisioning as well, the call home call for example) and there is an error reading the file show an error message. If a user does need the certificate data, we don't need to make this a requirement to render the template.

Reply:
@stejskalleos convinced me so +1 to have something like `<%= foreman_ssl_certificate %>` that fails in case it can't be read. We can start enforcing HTTPS.